### PR TITLE
Feature/near 16

### PR DESCRIPTION
**Pull request by irene-bbox**
Design and develop a brand new credit scoring algorithm for the _Covalent oracle_: an oracle for NEAR Protocol, which allows a user to authenticate through MetaMask and calculate a credit score based on the financial health of their ETH-based tokens held in their non-custodial MetaMask wallet.

### CODE REVIEW

File: support/metrics_covalent.py [hunk 1]
```diff
@@ -0,0 +1,652 @@
...
+
+    Parameters:
+        txn (dict): Covalent class A endpoint 'transactions_v2'
+        feedback (dict): score feedback
+
+    Returns:
+        txn (dict): formatted txn data containing only successful and non-dusty transactions
+    '''
+    try: 
+        if txn['quote_currency'] == 'USD':
```
Comment (m3mayllon) [inline support/metrics_covalent.py:23]: @irene-bbox what do you think about this?
`if txn['quote_currency'] == 'USD':`
`txn['items'] = [t for t in txn['items'] if t['successful'] and t['value_quote'] > 0]`

File: support/helper.py [hunk 2]
```diff
@@ -11,6 +11,7 @@ def head_tail_list(lst):
 
 
 def aggregate_currencies(ccy1, ccy2, fiats):
+    ccy1 = {k:v[1] for (k, v) in ccy1.items()}
```
Comment (m3mayllon) [inline support/helper.py:14]: @irene-bbox what are the impacts on coinbase model? same as the comments on [https://github.com/BalloonBox-Inc/NEARoracle-Oracle/pull/34/files#diff-ac46890f94fd4daea75b9542f384662762d11cb51efaee585c257eafa2fa294c](url)

**Review by m3mayllon — CHANGES_REQUESTED** [review]
@irene-bbox The model really looks nice. Well done!
Please just take a look at the comments I made, they're very little things.

File: market/coinmarketcap.py [hunk 3]
```diff
@@ -30,14 +30,41 @@ def coinmarketcap_currencies(api_key, limit):
         r = requests.get(url, headers=headers, params=params).json()
 
         top_currencies = dict(
-            [(n['symbol'], n['quote']['USD']['price']) for n in r['data']])
+            [(n['symbol'], (n['cmc_rank'], n['quote']['USD']['price'])) for n in r['data']])
```
[the remark this answers is not in the document]
Comment (irene-bbox) [reply market/coinmarketcap.py:33]: Nice catch! The function `coinmarketcap_currencies` is used once in the Coinbase model. Specifically, the output of `coinmarketcap_currencies` is one of the input parameters to another function called `aggregate_currencies` in the _helper.py_ file. To preserve the Coinbase model intact, I've added a line of code inside of `aggregate_currencies` to format the input parameter received from `coinmarketcap_currencies`. From now on, whenever we fetch top cryptos from Coinmarketcap, the function `coinmarketcap_currencies` will map a **tuple(int, float)** to a key (whereas initially it simply mapped a float to a key).

File: support/helper.py (hunk 2 above)
Comment (irene-bbox) [reply support/helper.py:14]: Thank you @m3mayllon ! Resolved. See [here](https://github.com/BalloonBox-Inc/NEARoracle-Oracle/pull/34/files#r913225865).

File: support/metrics_covalent.py (hunk 1 above)
Comment (irene-bbox) [reply support/metrics_covalent.py:23]: Brilliant! I've implemented the list comprehension as you recommended!

**Comment by irene-bbox** [comment]
Thanks @m3mayllon ! I agree with all your comments, implemented them, and pushed again. All is ready to merge now!

**Pull request merged**